packages/react-server/src/ReactServerStreamConfigBrowser.js

```diff
@@ -21,24 +21,64 @@ export function flushBuffered(destination: Destination) {
   // transform streams. https://github.com/whatwg/streams/issues/960
 }
 
+let currentView = null;
+let writtenBytes = 0;
+
 export function beginWriting(destination: Destination) {}
 
 export function writeChunk(
   destination: Destination,
   chunk: PrecomputedChunk | Chunk,
 ): void {
-  destination.enqueue(chunk);
+  if (currentView === null) {
+    currentView = new Uint8Array(512);
+    writtenBytes = 0;
+  }
+
+  if (chunk.length > currentView.length) {
```

Comment:
You can hard code this to 512. This gets important if we start using byobRequest since the current view could be smaller in that case. However, what really matters is whether it'll overflow *the next view*.

Reply:
Ah yup that makes sense